advisor/src/main/kotlin/advisors/VulnerableCode.kt

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.advisor.advisors
+
+import java.io.IOException
+import java.net.URI
+import java.time.Instant
+
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.coroutineScope
+
+import org.ossreviewtoolkit.advisor.AbstractAdvisorFactory
+import org.ossreviewtoolkit.advisor.Advisor
+import org.ossreviewtoolkit.clients.vulnerablecode.VulnerableCodeService
+import org.ossreviewtoolkit.model.AdvisorDetails
+import org.ossreviewtoolkit.model.AdvisorResult
+import org.ossreviewtoolkit.model.AdvisorSummary
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.Vulnerability
+import org.ossreviewtoolkit.model.config.AdvisorConfiguration
+import org.ossreviewtoolkit.model.config.VulnerableCodeConfiguration
+import org.ossreviewtoolkit.model.createAndLogIssue
+import org.ossreviewtoolkit.model.utils.toPurl
+import org.ossreviewtoolkit.utils.collectMessagesAsString
+import org.ossreviewtoolkit.utils.showStackTrace
+
+/**
+ * An [Advisor] implementation that obtains security vulnerability information from a
+ * [VulnerableCode][https://github.com/nexB/vulnerablecode] instance.
+ */
+class VulnerableCode(name: String, config: AdvisorConfiguration) : Advisor(name, config) {
+    class Factory : AbstractAdvisorFactory<VulnerableCode>("VulnerableCode") {
+        override fun create(config: AdvisorConfiguration) = VulnerableCode(advisorName, config)
+    }
+
+    companion object {
+        /** The number of elements to request at once in a bulk request. */
+        private const val BULK_FETCH_SIZE = 100
```

Comment:
How was this value determined? Is it even necessary here? Are there some docs that we could link?

Reply:
It is a wild guess (and copied from the NexusIQ advisor). Maybe we can get some qualified information when the bulk API is final on side of VulnerableCode.

Reply:
Wild-guessing is perfectly fine, but we should document it then.